addons/binding/org.openhab.binding.dlinkupnpcamera/src/main/java/org/openhab/binding/dlinkupnpcamera/handler/DlinkUpnpCameraHandler.java

```diff
@@ -0,0 +1,327 @@
+import java.net.MalformedURLException;
+import java.net.PasswordAuthentication;
+import java.net.URL;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import org.eclipse.smarthome.config.discovery.DiscoveryServiceRegistry;
+import org.eclipse.smarthome.core.library.types.NextPreviousType;
+import org.eclipse.smarthome.core.library.types.PlayPauseType;
+import org.eclipse.smarthome.core.library.types.RawType;
+import org.eclipse.smarthome.core.thing.ChannelUID;
+import org.eclipse.smarthome.core.thing.Thing;
+import org.eclipse.smarthome.core.thing.ThingStatus;
+import org.eclipse.smarthome.core.thing.ThingStatusDetail;
+import org.eclipse.smarthome.core.thing.binding.BaseThingHandler;
+import org.eclipse.smarthome.core.types.Command;
+import org.eclipse.smarthome.core.types.RefreshType;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOParticipant;
+import org.eclipse.smarthome.io.transport.upnp.UpnpIOService;
+import org.openhab.binding.dlinkupnpcamera.config.DlinkUpnpCameraConfiguration;
+import org.openhab.io.net.actions.Ping;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * The {@link DlinkUpnpCameraHandler} is responsible for handling commands, which are
+ * sent to one of the channels.
+ *
+ * @author Yacine Ndiaye
+ * @author Antoine Blanc
+ * @author Christopher Law
+ */
+public class DlinkUpnpCameraHandler extends BaseThingHandler implements UpnpIOParticipant {
+
+    private static final int DEFAULT_REFRESH_INTERVAL = 30;
+    private static final int PING_TIMEOUT = 20;
+
+    private String hostname;
+
+    private Logger logger = LoggerFactory.getLogger(DlinkUpnpCameraHandler.class);
+
+    private UpnpIOService service;
+    private ScheduledFuture<?> pollingJob;
+
+    private Runnable pollingRunnable = new Runnable() {
+
+        @Override
+        public void run() {
+            try {
+                logger.debug("Polling...");
+
+                // First check if the camera is set in the UPnP service registry
+                // If not, set the thing state to OFFLINE and wait for the next poll
+                if (!isUpnpDeviceRegistered()) {
+                    logger.debug("UPnP device {} not yet registered", getUDN());
+                    updateStatus(ThingStatus.OFFLINE);
+                    return;
+                }
+
+                // Check if the camera can be joined
+                // If not, set the thing state to OFFLINE and do nothing else
+                updatePlayerState();
+                if (getThing().getStatus() != ThingStatus.ONLINE) {
+                    return;
+                }
+            } catch (Exception e) {
+                logger.debug("Exception during poll : {}", e);
+            }
+        }
+    };
+
+    @Override
+    public void initialize() {
+
+        logger.debug("initializing handler for thing {}", getThing().getUID());
+
+        if (getUDN() != null) {
+            updateStatus(ThingStatus.ONLINE);
```

Comment:
You set the thing to ONLINE here although you don't have any clue whether you can successfully communicate with it or not. Apart from that, `super.initialize()` does the exact same thing. It's perfectly fine if you don't know that yet. You may actually either leave it to the polling job to set the first state (but then you must **guarantee** that this eventually **will** happen. Alternatively you can set the thing to UNKNOWN state here, which is safer to do and I would recommend it.